### ISSUE TITLE
[A11y-Defect-1]  ⚠️  Buttons cut off on mobile for in-progress confirmation modal

### ISSUE THREAD

**Issue by eileen-coforma** [department-of-veterans-affairs/va.gov-team#59774]
## Current behavior
The yes and no buttons are cut off on the right edge on the mobile modal when a user tries to start a form over.
<details><summary>Screenshot</summary>

![image.png](https://images.zenhubusercontent.com/630d073e647e629355675b4b/d764be42-b82d-42a7-86e3-b1783ee63e8c)
</details>

## Expected behavior
Buttons should show full width to the modal, and resize based on screen size.

## Background context
Found this when testing the Request a Board Appeal form on staging.

## Acceptance Criteria
- [ ] Confirmation modal shows button entirely on mobile
- [ ] Confirmation modal buttons are accessible to VoiceOver iOS users

## Tasks

- [ ] Fix the button width issue for mobile
- [ ] Check on staging with a mobile device
- [ ] Investigate the reason why VoiceOver iOS is unable to swipe to the confirmation modal buttons
- [ ] Test the confirmation modal buttons on other mobile screen reader technologies
- [ ] Fix the accessibility issue
- [ ] Confirm it has been fixed on all major screen reader and browser combinations

## Definition of done

- [ ] All tasks complete

**Comment by saderagsdale** [comment]
@eileen-coforma post in platform support channel.

**Comment by jaredcunha** [comment]
This issue is much more pronounced on screen sizes slightly wider than standard mobile.
<img width="475" alt="Screenshot 2023-06-09 at 2 02 47 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/3c6b526f-d669-4780-aa61-d1cf5ac2aa26">

One issue is that the buttons have a fixed width.
<img width="734" alt="Screenshot 2023-06-09 at 2 03 31 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/ec415996-ae83-45d4-b4ba-97e78bee8182">

[CSS file for reference](https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/components/va-button-pair/va-button-pair.scss)

My guess is that the [button pair component](https://design.va.gov/components/button/button-pair) is a draft component and has been tested on a mobile device in the middle of a page, but not in a modal window, which is much more constrained on the left and right side.

### I think there might be a couple of options to address:

1.  **(Platform)** Update the component to use percentage-based widths (100%), and use flexbox to handle the lockups (and might need to add `width: 100%` to the container. See inline style for `<va-button-pair>` and the `width: 100%` on on `<va-button>`.
<img width="1428" alt="Screenshot 2023-06-09 at 2 17 14 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/a51b230d-ef4b-4ca5-bc89-a61575263676">

I feel like this is the ideal fix. Button pairs are common in modals, and I think this needs to be done in order to graduate the web component.

2. **(In the app)** Do not use the `<va-button-pair>` component, and instead, just write custom styles for organizing two buttons in the modal.

**Comment by joshkimux** [comment]
Adding to this conversation, this adds a critical issue for iOS VoiceOVer users who are unable to swipe to proceed to the buttons.

https://images.zenhubusercontent.com/133843125/3369e737-deac-4e32-8832-415d54bdb3cb/modal_bug_no_access_to_buttons.mp4

**Comment by humancompanion-usds** [comment]
This may be related to the fix to [this issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1798).
